lang/cs/Org.Apache.REEF.Common/Telemetry/MetricsService.cs

```diff
@@ -34,14 +34,35 @@ namespace Org.Apache.REEF.Common.Telemetry
     internal sealed class MetricsService : IObserver<IContextMessage>
     {
         private static readonly Logger Logger = Logger.GetLogger(typeof(MetricsService));
-        private readonly IDictionary<string, ICounter> _counters = new ConcurrentDictionary<string, ICounter>();
+
+        /// <summary>
+        /// Contains Counters received in the Metrics service
+        /// </summary>
+        private readonly CountersData _countersData;
+
+        /// <summary>
+        /// A set of metrics sinks
+        /// </summary>
+        private readonly ISet<IMetricsSink> _metricsSinks;
```

Comment:
We have only one MetricsService which contains the collection of IMetricsSink. It is readonly and is initialized in the constructor of MetricsService so there is not chance to be updated.  We inject IMetricsSink through named parameter which is an ISet of IMetricsSink, this is similar to we inject handlers for driver. Also Tang supports ISet only for this type of named parameters.